backend/src/permissions.ts

```diff
@@ -0,0 +1,37 @@
+import { rule, shield } from "graphql-shield";
+import PostDatasource from "./datasources/post.datasource";
+import { JwtPayload } from "./jwt-payload";
+
+const isAuthenticated = rule({ cache: "contextual" })(
+  async (parent, args, ctx) => {
+    return ctx.user !== null;
```

Comment:
```suggestion
    return ctx.user !== null && ctx.user !== undefined;
```
user could be undefined